handlers/src/main/java/com/nextdoor/bender/handler/s3/S3EventIterator.java

```diff
@@ -187,6 +190,11 @@ private void updateCursor() {
       logger.trace("s3 get request id: " + client.getCachedResponseMetadata(req).getRequestId()
           + " host: " + client.getCachedResponseMetadata(req).getHostId() + " cloudfrontid: "
           + client.getCachedResponseMetadata(req).getCloudFrontId());
+      Date objOpenTime = new Date();
+      long spoutLag = obj.getObjectMetadata().getLastModified().getTime() - objOpenTime.getTime();
+      if(spoutLag > ALLOWABLE_SPOUT_LAG_IN_MS) {
+        logger.debug("late file (" + spoutLag + " ms) s3://" + bucketName + "/" + key);
```

Comment:
This is notifying us that the SNS notification about this file took a long time to receive, correct? If so, lets re-phrase this:

`Notification for ${file} was received at ${snsNotificationTimestamp} - ${x} seconds after the file landed in S3 (${fileCreationTimestamp})`.

Reply:
👍 on Matt's comment. Also, if you use "spout" here please throw in a comment of what it is measuring (between what and what). Kind of like @diranged 's example.

Reply:
On second thought, we should figure out how to put make this delta the spout lag metric.

Reply:
I have updated this PR to track notification delay instead of spout lag as we already have an existing metric and really at this point we are only concerned with logging an anomalous underlying event rather than measuring it on an ongoing basis.